doc-content/drools-docs/src/main/asciidoc/DecisionEngine/cep-modes-con.adoc

```diff
@@ -0,0 +1,69 @@
+[id='cep-modes-con_{context}']
+
+= Event processing modes in the {DECISION_ENGINE}
+
+The {DECISION_ENGINE} runs in either cloud mode or stream mode. In cloud mode, the {DECISION_ENGINE} processes facts as facts with no temporal constraints, independent of time, and in no particular order. In stream mode, the {DECISION_ENGINE} processes facts as events with strong temporal constraints, in real time or near real time. Stream mode uses synchronization to make event processing possible in {PRODUCT}.
+
+Cloud mode::
+Cloud mode is the default operating mode of the {DECISION_ENGINE}. In cloud mode, the {DECISION_ENGINE} uses a many-to-many pattern-matching algorithm, which treats the events as an unordered cloud. Events still have time stamps, but the {DECISION_ENGINE} running in cloud mode cannot draw relevance from the time stamp because cloud mode ignores the present time. This mode uses the rule constraints to find the matching tuples to activate and execute rules.
++
```

Comment:
I don't understand here with "many-to-many" pattern matching. I'd remove it.

Reply:
Haha, me neither actually, on second thought. I thought it was a Drools concept :) Done. I removed it and changed to this:

>In cloud mode, the {DECISION_ENGINE} treats events as an unordered cloud.